cmd/diagnostic/main.go

```diff
@@ -64,14 +65,20 @@ func main() {
 	var networkPeers map[string]string
 	var joinedNetwork bool
 	if *networkPtr != "" {
-		logrus.Infof("Joining the network:%s", *networkPtr)
-		resp, err = http.Get(fmt.Sprintf(joinNetwork, *ipPtr, *portPtr, *networkPtr))
-		if err != nil {
-			logrus.WithError(err).Fatalf("Failed joining the network")
+		if *joinPtr {
+			logrus.Infof("Joining the network:%q", *networkPtr)
+			resp, err = http.Get(fmt.Sprintf(joinNetwork, *ipPtr, *portPtr, *networkPtr))
+			if err != nil {
+				logrus.WithError(err).Fatalf("Failed joining the network")
+			}
+			httpIsOk(resp.Body)
+			joinedNetwork = true
 		}
-		httpIsOk(resp.Body)
+
 		networkPeers = fetchNodePeers(*ipPtr, *portPtr, *networkPtr)
-		joinedNetwork = true
+		if len(networkPeers) == 0 {
+			logrus.Warnf("There is no peer on network %q, check the network ID, and verify that is the non truncated version", *networkPtr)
```

Comment:
Is it safe to continue here rather than exit with fatalf?

Reply:
yep it is, the output will only show no peers on the network and the validation with the entries will show a warning like:
```
WARN[0000] The element with key:23591a2e3795dc36c994eae6074239c91a9e647784af374cea1abf33373acb07 does not belong to any node on this network                                                                                                                                    
```

Reply:
ok .. thanks!